docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,483 @@
+
+[source, shell, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vertx"
+----
+
+If you have an already created project, the `vertx` extension can be added to an existing {project-name} project with
+the `add-extension` command:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="vertx"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-vertx</artifactId>
+</dependency>
+----
+
+== Accessing Vert.x
+
+Once the extension added, you can access the _managed_ Vert.x instance using `@Inject`:
+
+[source, java]
+----
+@Inject Vertx vertx;
+----
+
+If you are familiar with Vert.x, you know that Vert.x provides different API models.
+For instance _bare_ Vert.x uses callbacks, the RX Java 2 version uses `Single`, `Maybe`, `Completable`, `Observable` and `Flowable`.
+
+{project-name} provides 3 Vert.x APIs:
+
+[options="header"]
+|===
+
+| Name | Code | Description
+
+| _bare_ | `@Inject io.vertx.core.Vertx vertx` | _bare_ Vert.x instance, the API uses callbacks.
+
+| RX Java 2 | `@Inject io.vertx.reactivex.core.Vertx vertx` | RX Java 2 Vert.x, the API uses RX Java 2 types.
+
+| _Axle_ | `@Inject io.vertx.axle.core.Vertx vertx` | Axle Vert.x, the API uses `CompletionStage` and `Reactive Streams`.
+
+|===
+
+TIP: You may inject any of the 3 flavors of `Vertx` as well as the `EventBus` in your {project-name} application beans: `bare`, `Axle`, `RxJava2`.
+They are just shims and rely on a single _managed_ Vert.x instance.
+
+You will pick one or the other depending on your use cases.
+
+- `bare`: for advanced usage or if you have existing Vert.x code you want to reuse in your {project-name} application
+- `Axle`: works well with {project-name} and MicroProfile APIs (`CompletionStage` for single results and `Publisher` for streams)
+- `Rx Java 2`: when you need support for a wide range of data transformation operators on your streams
+
+The following snippets illustrate the difference between these 3 APIs:
+
+[source, java]
+----
+// Bare Vert.x:
+vertx.fileSystem().readFile("lorem-ipsum.txt", ar -> {
+    if (ar.succeeded()) {
+        System.out.println("Content:" + ar.result().toString("UTF-8"));
+    } else {
+        System.out.println("Cannot read the file: " + ar.cause().getMessage());
+    }
+});
+
+// Rx Java 2 Vert.x
+vertx.fileSystem().rxReadFile("lorem-ipsum.txt")
+    .map(buffer -> buffer.toString("UTF-8"))
+    .subscribe(
+            content -> System.out.println("Content: " + content),
+            err -> System.out.println("Cannot read the file: " + err.getMessage())
+    );
+
+// Axle API:
+vertx.fileSystem().readFile("lorem-ipsum.txt")
+    .thenApply(buffer -> "Content: " + buffer.toString("UTF-8"))
```

Comment:
You're cheating here: it does not implement things like the Rx version above, they should be much more similar.

Reply:
Replaced with `whenComplete`. However, each model comes with its own idioms and I generally change a bit the processing depending on the API I have.